packages/taco/src/conditions/base/index.ts

```diff
@@ -1,37 +1,70 @@
 import {
   CompoundConditionProps,
   compoundConditionSchema,
+  CompoundConditionType,
 } from '../compound-condition';
-import { Condition } from '../condition';
+import { Condition, ConditionProps } from '../condition';
 
-import { ContractConditionProps, contractConditionSchema } from './contract';
-import { RpcConditionProps, rpcConditionSchema } from './rpc';
-import { TimeConditionProps, timeConditionSchema } from './time';
+import { ContractConditionProps, contractConditionSchema, ContractConditionType } from './contract';
+import { RpcConditionProps, rpcConditionSchema, RpcConditionType } from './rpc';
+import { TimeConditionProps, timeConditionSchema, TimeConditionType } from './time';
+
+type OmitType<T> = Omit<T, 'conditionType'>
```

Comment:
`OmitType<...>` reads a little odd for me, like "omit the type that's being provided to it" almost eg. OmitType<Foo> almost reads like "omit the Foo type" (to me).

What do you think about `OmitConditionType<...>` or `OmitConditionTypeFrom<...>`? or something like that?

Reply:
I thought it was a bit wordy, but it make sense